File: backend/models/DanhMuc.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const DanhMuc = sequelize.define(
  "DanhMuc",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    ten_danh_muc: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    slug: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    danh_muc_cha_id: {
      type: DataTypes.INTEGER,
      allowNull: true,
    },
    thu_tu: {
      type: DataTypes.INTEGER,
      allowNull: true,
      defaultValue: 0,
    },
    hinh_anh: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    mo_ta: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    trang_thai: {
      type: DataTypes.STRING(20),
      allowNull: false,
      defaultValue: "active",
    },
    hien_thi_sidebar: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
    },
  },
  {
    tableName: "DanhMuc",
    timestamps: false,
  },
);

module.exports = DanhMuc;

File: backend-csharp/Models/DanhMuc.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class DanhMuc
{
    public long Id { get; set; }

    public string TenDanhMuc { get; set; } = null!;

    public string Slug { get; set; } = null!;

    public long? DanhMucChaId { get; set; }

    public string? HinhAnh { get; set; }

    public string? MoTa { get; set; }

    public int ThuTu { get; set; }

    public string TrangThai { get; set; } = null!;

    public bool? HienThiSidebar { get; set; }

    public virtual DanhMuc? DanhMucCha { get; set; }

    public virtual ICollection<DanhMuc> InverseDanhMucCha { get; set; } = new List<DanhMuc>();

    public virtual ICollection<SanPham> SanPhams { get; set; } = new List<SanPham>();
}
